### ISSUE TITLE
Bitmap too large to be uploaded into a texture

### ISSUE THREAD

**Issue by gnuyork** [ArthurHub/Android-Image-Cropper#210]
`CropImageView.setImageUriAsync((Uri) getIntent().getExtras().get(IMAGE_URI));`

fails on my old Samsung Galaxy Nexus with the following log output and no Exception:

> W/OpenGLRenderer:  Bitmap too large to be uploaded into a texture (1944x2592, max=2048x2048)

As a result the ImageView remains empty (white). 
The image is taken is taken by the device's camera.

It would be nice to have a getter for the ImageView inside CropImageView. So I could fix this on my own by loading the image with Glide, which I also use in my app.

**Comment by ArthurHub** [comment]
First, verify you are using the latest version as I have code that should handle this scenario.
If so, you will have to debug [BitmapUtils.getMaxTextureSize()](https://github.com/ArthurHub/Android-Image-Cropper/blob/master/cropper/src/main/java/com/theartofdev/edmodo/cropper/BitmapUtils.java#L644) to understand why it doesn't return the correct value. This is the method that calculates the max allowed size for texture rendering.
Let me know what you find out.

p.s. how will Glide help you for camera captured image?

**Comment by gnuyork** [comment]
I'm using version 2.3.+, so it should be 2.3.1 now.

It seems like Glide handles to resize the images before loading them into ImageViews. The error does not occur when with Glide.

The strange thing is that the error does not occur everytime as I figured out now. In about one of ten tries to load a fresh camera picture, it works fine. It also works for already stored camera images with the same resolution. 

That's really strange. Could also be a CyanogenMod bug. I'll try to figure it out.

Could you consider to add a getter for the ImageView?

**Comment by ArthurHub** [comment]
Exposing

**Comment by ArthurHub** [comment]
Exposing the ImageView is problematic, many potential bugs and issues.
Sounds very strange... possible bug in CyanogenMod

**Comment by gnuyork** [comment]
Running BitmapUtils.getMaxTextureSize() returns 2048 which is fine. I'll post as soon as I know what the reason is.

**Comment by gnuyork** [comment]
I've fixed the bug! :) Good news for you, ArthurHub: In fact it was not related to the Image Cropper. 

So what was the problem?
I've created the camera output file using File.createTempFile() and passed it to the camera app. That was a bad idea! I figured out that it's better to pass just a file name of a non-existing file to the camera app, so that it has to create the file on it's own. 
It seems like the camera app returns before the output file has been written. With the already created temp file the Image Cropper tried to load an empty file into the ImageView. For some reason trying to do this creates the OpenGL error above. That's very confusing. Maybe Android tries to read the image's width/height and returns Integer.MAX_VALUE on failure? Dunno.
After removing the createTempFile() code and just passing a fresh file name the problem was solved.

Thank you for trying to help and also thanks for this really nice project!
